robot-code/micro-python/sumobot.py

```diff
@@ -110,4 +108,31 @@ def drive(self, speed):
             self.pin2.duty(duty_cycle)
         else:
             self.pin1.duty(0)
-            self.pin2.duty(0)
\ No newline at end of file
+
+class Sensor:
+    ### Sets up a sensor given pin number and sensor type
+    ### Does not return
+    def __init__(self, pin_number, pin_type):
+        assert pin_type=="analog" or pin_type=="digital"
```

Comment:
This is not a very pythonic way to do this. 3 issues:
1) You are using this assert to limit inputs. The user won't know what the valid inputs are until they hit the assert. you should think about using an enum so that the user has some hint as to what the valid inputs are.
2) Asserts are for testing not production. They will be stripped with a -O flag. You should raise a value exception instead.
3) Asserts should have messages to explain what the failure is. Even if you keep this there should be a string explaining what's going on when they hit this